simple/cm_data.c

```diff
@@ -382,8 +386,28 @@ static int run(void)
 		} else {
 			ret = FI_SUCCESS;
 		}
+	} else {
+		ret = ft_sock_listen(sock_service);
+		if (ret)
+			goto err2;
+		ret = ft_sock_accept();
+		if (ret)
+			goto err1;
+
+		for (i = 1; i <= cm_data_size; i <<= 1) {
+			printf("trying with data size: %zu\n", i);
+			ret = server(i);
+			if (ret)
+				goto err1;
+
+			ret = ft_sock_sync(0);
+			if (ret)
+				goto err1;
+		}
```

Comment:
Minor suggestion: We can keep it as a single loop, and with the connect/accept calls before the loop. That we we don't need to have the loop code repeated.